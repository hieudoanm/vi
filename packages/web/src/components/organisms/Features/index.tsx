import { Container } from '@mui/material';

export type Feature = { title: string; description: string };

export type FeaturesProps = {
  id: string;
  title: string;
  subtitle: string;
  features: Feature[];
};

const Features: React.FC<FeaturesProps> = ({
  id = '',
  title = '',
  subtitle = '',
  features = [],
}) => {
  return (
    <section id={id} className="py-20">
      <Container>
        <p className="text-center mb-3 text-blue-500 font-medium">{subtitle}</p>
        <h1 className="text-center mx-auto mb-8 text-4xl font-bold">{title}</h1>
        <div className="max-w-2xl mx-auto">
          <div className="grid grid-cols-1 md:grid-cols-2 gap-10">
            {features.map(({ title, description }: Feature) => {
              return (
                <div
                  key={`feature-${title.toLowerCase()}`}
                  className="flex flex-col justify-center items-center my-8"
                >
                  <h2 className="mb-6 text-3xl font-medium">{title}</h2>
                  <p className="text-center">{description}</p>
                </div>
              );
            })}
          </div>
        </div>
      </Container>
    </section>
  );
};

Features.displayName = 'Features';
Features.defaultProps = { features: [] };

export default Features;
