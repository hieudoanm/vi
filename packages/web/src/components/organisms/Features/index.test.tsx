import { render } from '@testing-library/react';
import Features from '.';

describe('Features', () => {
  test('render default', () => {
    const wrapper = render(
      <Features id="id" title="title" subtitle="subtitle" features={[]} />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
