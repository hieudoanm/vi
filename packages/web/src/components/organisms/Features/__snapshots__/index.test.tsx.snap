// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features render default 1`] = `
<div>
  <section
    class="py-20"
    id="id"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <p
        class="text-center mb-3 text-blue-500 font-medium"
      >
        subtitle
      </p>
      <h1
        class="text-center mx-auto mb-8 text-4xl font-bold"
      >
        title
      </h1>
      <div
        class="max-w-2xl mx-auto"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 gap-10"
        />
      </div>
    </div>
  </section>
</div>
`;
